pkg/kubectl/cmd/label.go

```diff
@@ -57,6 +57,9 @@ $ kubectl label pods foo bar-`
 )
 
 func NewCmdLabel(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	p := kubectl.NewHumanReadablePrinter(false, false, false, false, []string{})
```

Comment:
This is not correct - you need to be getting all the printers via the factory.  Otherwise, it's impossible for someone via the factory to provide their own human readable printers.  If there isn't an appropriate method on the factory, we need to add it.